library/node.c

```diff
@@ -345,8 +395,11 @@ int dnet_idc_update_backend(struct dnet_net_state *st, struct dnet_backend_ids *
 	idc->st = st;
 	idc->group = g;
 	idc->backend_id = backend->backend_id;
+	idc->weight = DNET_STATE_DEFAULT_WEIGHT;
 
-	list_add_tail(&idc->state_entry, &st->idc_list);
+	pthread_rwlock_wrlock(&st->idc_lock);
+	dnet_idc_insert(st, idc);
```

Comment:
Please, use **_nolock** at the end of a function name if this function requires mutex to be locked.